pkg/osquery/table/chrome_login_keychain_emails.go

```diff
@@ -0,0 +1,140 @@
+}
+
+func ChromeLoginKeychainEmails(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	c := &ChromeLoginKeychainEmailsTable{
+		client: client,
+		logger: logger,
+	}
+	columns := []table.ColumnDefinition{
+		table.TextColumn("username"),
+		table.TextColumn("email"),
+		table.BigIntColumn("count"),
+	}
+	return table.NewPlugin("kolide_chrome_login_keychain_emails", columns, c.generate)
+}
+
+type ChromeLoginKeychainEmailsTable struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+}
+
+func (c *ChromeLoginKeychainEmailsTable) generateForPath(ctx context.Context, path string, username string) ([]map[string]string, error) {
+	dir, err := ioutil.TempDir("", "kolide_chrome_login_keychain_emails")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	dst := filepath.Join(dir, "tmpfile")
+	if err := fs.CopyFile(path, dst); err != nil {
+		return nil, err
+	}
+
+	db, err := sql.Open("sqlite3", dst)
+	if err != nil {
+		return nil, err
+	}
+	defer db.Close()
+
+	db.Exec("PRAGMA journal_mode=WAL;")
+
+	rows, err := db.Query("SELECT username_value, count(*) AS count FROM logins GROUP BY lower(username_value)")
+	if err != nil {
+		return nil, errors.Wrap(err, "query rows from chrome login keychain db")
+	}
+	defer rows.Close()
+
+	var results []map[string]string
+
+	// loop through all the sqlite rows and add them as osquery rows in the results map
+	for rows.Next() { // we initialize these variables for every row, that way we don't have data from the previous iteration
+		var username_value string
+		var username_count string
+		if err := rows.Scan(&username_value, &username_count); err != nil {
+			return nil, errors.Wrap(err, "scanning chrome login keychain db row")
+		}
+		// append anything that could be an email
+		if strings.Contains(username_value, "@") {
+			results = append(results, map[string]string{
+				"username": username,
+				"email":    username_value,
+				"count":    username_count,
+			})
+		}
+	}
+	return results, nil
+}
+
+func (c *ChromeLoginKeychainEmailsTable) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var usernames []string
+	q, ok := queryContext.Constraints["username"]
+	if ok && len(q.Constraints) != 0 {
+		for _, constraint := range q.Constraints {
+			usernames = append(usernames, constraint.Expression)
+		}
+	} else {
+		return nil, errors.New("The kolide_chrome_login_keychain table requires that you specify a constraint WHERE username =")
+	}
+
+	var results []map[string]string
+	osProfileDirs, ok := profileDirs[runtime.GOOS]
+	if !ok {
+		osProfileDirs, ok = profileDirs["default"]
+		if !ok {
+			return results, errors.New("No profileDir for this platform")
+		}
+	}
+
+	for _, username := range usernames {
+		for _, profileDir := range osProfileDirs {
+			userPaths, err := findFileInUserDirs(filepath.Join(profileDir, "Default/Login Data"), WithUsername(username))
```

Comment:
@blaedj is right. We should be looking in all the profiles, not just `Default`. For example, I don't have a `Default` but I do have 7 profiles.

Probably enumerate the directories here, and check anything that starts with Default or Profile